scipy/optimize/_remove_redundancy.py

```diff
@@ -181,7 +186,8 @@ def _remove_redundancy_dense(A, rhs):
     # mistakes and ambiguities - which is strange, because the rest of
     # Andersen's papers are quite good.)
 
-    B = A[:, b]
+    lu = np.eye(m, order='F'), np.arange(m)  # LU for initial basis is trivial
```

Comment:
I guess I still want to use `eye` here. @Kai-Striega would you prefer I do `zeros`+`fill_diagonal`?

Reply:
You can also use `np.eye(m).T` to get fortran order.

Reply:
Thanks. Would it behave identically, though?
I went ahead and replaced with `np.zeros` + `np.fill_diagonal`, as it's just as fast and a little more explicit about the Fortran order.

Reply:
>Would it behave identically, though?

The identity matrix is symmetric so it is still an identity after transposing. Further, `.T` just creates a strided view. You can try for yourself:
```python
>>> x = np.eye(4).T
>>> x
array([[1., 0., 0., 0.],
       [0., 1., 0., 0.],
       [0., 0., 1., 0.],
       [0., 0., 0., 1.]])
>>> x.flags
  C_CONTIGUOUS : False
  F_CONTIGUOUS : True
  OWNDATA : False
  WRITEABLE : True
  ALIGNED : True
  WRITEBACKIFCOPY : False
  UPDATEIFCOPY : False
```

Reply:
I know, and I saw the flags. The only one that was different was 'OWNDATA', which was expected, as I knew it would be a view. Just confirming, as I don't know everything under the hood. Thanks.

Reply:
>  @Kai-Striega would you prefer I do zeros+fill_diagonal?

I'm happy with either. zeros+fill_diagonal was just a suggestion as the (then) existing code fails the test suite.

Reply:
Haha yeah sounds like a good reason to change it.

Reply:
either way, `np.eye(m, order='F')` is still the most legible option (in my opinion). It would probably be worth adding a comment why it was *not* used and to update it when the NumPy version is increased.

Reply:
Oops, yeah, I meant to do that.

Reply:
I should have just waited : ) The min NumPy is now 1.14.5.